lib/internal/util/inspect.js

```diff
@@ -861,10 +862,29 @@ function formatError(err, constructor, tag, ctx) {
       }
     }
   }
+  // Ignore the error message if it's contained in the stack.
+  let pos = err.message && stack.indexOf(err.message) || -1;
+  if (pos !== -1)
+    pos += err.message.length;
   // Wrap the error in brackets in case it has no stack trace.
-  const stackStart = stack.indexOf('\n    at');
+  const stackStart = stack.indexOf('\n    at', pos);
   if (stackStart === -1) {
     stack = `[${stack}]`;
+  } else if (ctx.colors) {
+    // Highlight userland code and node modules.
+    let newStack = stack.slice(0, stackStart);
+    const lines = stack.slice(stackStart + 1).split('\n');
+    for (const line of lines) {
+      if (/[/\\]node_modules[/\\]/.test(line)) {
+        newStack += `\n${ctx.stylize(line, 'module')}`;
+      // Node.js core files do not begin with an slash.
+      } else if (/    at .+ \([/\\].+:\d+:\d+\)$/u.test(line)) {
```

Comment:
do we have a "is internal module" API or something we can use? this seems brittle.

Reply:
No, there's none (that I am aware of) and such API could only work for stack traces that have not been tampered with (but that should be fine since it would still be a significant reliability improvement over the current situation).

However, all Node.js versions I know should work with the here used regular expression and I doubt that it's an issue if it rarely fails to properly identify one frame.